### DISCUSSION TITLE
Generating Elasticsearch Nested Queries from UI Inputs

### DISCUSSION THREAD

**Discussion by sharopcha** [react-querybuilder/react-querybuilder#649] [Q&A] [answered]
Hello Community,

I'm working on a project where we need to allow users to generate Elasticsearch queries through a UI, specifically targeting nested fields within our Elasticsearch documents. Our challenge is finding an efficient way to transform user inputs from the UI into Elasticsearch nested queries.

Given a document structure with nested fields, such as:

```
{
  "user": {
    "name": "John Doe",
    "projects": [
      {
        "name": "Project Alpha",
        "description": "Description of Project Alpha"
      },
      {
        "name": "Project Beta",
        "description": "Description of Project Beta"
      }
    ]
  }
}
```

We need a way for users to construct queries (e.g., finding users with a project named "Alpha") and translate these into valid Elasticsearch nested queries.

example of expected query

```
{
  "query": {
    "nested": {
      "path": "user.projects",
      "query": {
        "bool": {
          "must": [
            {
              "match": {
                "user.projects.name": "Project Alpha"
              }
            }
          ]
        }
      }
    }
  }
}
```

**Comment by jakeboone02** [comment, marked as answer]
This library doesn't really have first-class support for nested fields, but I think [this one](https://github.com/ukrbublik/react-awesome-query-builder/) does.

I put some related ideas down in #515 but haven't acted on them yet.

**Reply by sharopcha** [reply]
Thank you!
